cmd/ci-operator/main.go

```diff
@@ -1495,3 +1483,49 @@ func getPullSecretFromFile(filename string) (*coreapi.Secret, error) {
 	secret.Data[coreapi.DockerConfigJsonKey] = src
 	return secret, nil
 }
+
+func getResolverInfo(jobSpec *api.JobSpec, o *options) *load.ResolverInfo {
+	// identify org, repo, and branch from refs
+	var refs prowapi.Refs
+	if jobSpec.Refs != nil {
+		refs = *jobSpec.Refs
+	}
+	info := &load.ResolverInfo{
+		Address: o.resolverAddress,
+		Org:     refs.Org,
+		Repo:    refs.Repo,
+		Branch:  refs.BaseRef,
+		Variant: o.variant,
+	}
+	if len(info.Org) > 0 && len(info.Repo) > 0 && len(info.Branch) > 0 {
+		return info
+	}
+
+	// if org, repo, or branch could not be identified, try extra refs
+	for _, ref := range jobSpec.ExtraRefs {
```

Comment:
This code allows `refs` to get filled from a mixture of different sources. IMHO if we filled any part of it from one source, we must return and then have a check if everything we need is filled, rather than starting to mix.

Also, please do empty checks for strings via `s == ""`, because that tells the reader of your code that this is a string.